src/compiler/command_line_parser.rs

```diff
@@ -1,6 +1,2980 @@
+use std::array::IntoIter;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::iter::FromIterator;
 use std::rc::Rc;
 
-use crate::{CompilerOptions, ParsedCommandLine};
+use crate::{
+    CommandLineOption, CommandLineOptionBase, CommandLineOptionInterface,
+    CommandLineOptionMapTypeValue, CommandLineOptionOfBooleanType, CommandLineOptionOfCustomType,
+    CommandLineOptionOfListType, CommandLineOptionOfNumberType, CommandLineOptionOfStringType,
+    CommandLineOptionType, CompilerOptions, Diagnostics, ImportsNotUsedAsValues, JsxEmit,
+    ModuleKind, ModuleResolutionKind, NewLineKind, ParsedCommandLine, ScriptTarget,
+    StringOrDiagnosticMessage, TsConfigOnlyOption,
+};
+
+thread_local! {
+    pub(crate) static compile_on_save_command_line_option: Rc<CommandLineOption> =
+        CommandLineOptionOfBooleanType::new(CommandLineOptionBase {
+            _command_line_option_wrapper: RefCell::new(None),
+            name: "compileOnSave".to_string(),
+            type_: CommandLineOptionType::Boolean,
+            is_file_path: None,
+            short_name: None,
+            description: None,
+            default_value_description: Some(StringOrDiagnosticMessage::String("false".to_string())),
+            param_type: None,
+            is_tsconfig_only: None,
+            is_command_line_only: None,
+            show_in_simplified_help_view: None,
+            category: None,
+            strict_flag: None,
+            affects_source_file: None,
+            affects_module_resolution: None,
+            affects_bind_diagnostics: None,
+            affects_semantic_diagnostics: None,
+            affects_emit: None,
+            affects_program_structure: None,
+            transpile_option_value: None,
+        })
+        .into();
+}
+
+thread_local! {
+    pub(crate) static jsx_option_map: HashMap<&'static str, JsxEmit> =
```

Comment:
This seemed to work fine, per a previous PR comment I should try and update the other static `HashMap`'s to use `&'static str`'s rather than allocating `String`'s for all of the entries